source/common/network/connection_impl.h

```diff
@@ -149,14 +149,14 @@ class ConnectionImpl : public ConnectionImplBase, public TransportSocketCallback
   StreamInfo::StreamInfo& stream_info_;
   FilterManagerImpl filter_manager_;
 
-  // Ensure that if the consumer of the data from this connection isn't
-  // consuming, that the connection eventually stops reading from the wire.
-  Buffer::WatermarkBuffer read_buffer_;
   // This must be a WatermarkBuffer, but as it is created by a factory the ConnectionImpl only has
   // a generic pointer.
   // It MUST be defined after the filter_manager_ as some filters may have callbacks that
   // write_buffer_ invokes during its clean up.
   Buffer::InstancePtr write_buffer_;
+  // Ensure that if the consumer of the data from this connection isn't
+  // consuming, that the connection eventually stops reading from the wire.
+  Buffer::InstancePtr read_buffer_;
```

Comment:
optional, for these new pointers, worth calling out they're never non-null, or overkill?

Reply:
Done.